scripts/setup_rdma_env.sh

```diff
@@ -0,0 +1,135 @@
+    modprobe irdma
+}
+
+configure_irdma() {
+    echo "Configuring irdma driver..."
+
+    # enable RoCE
+    roce_ena_val=$(grep "options irdma roce_ena=" /etc/modprobe.d/irdma.conf | cut -d "=" -f 2)
+    if [[ -z "$roce_ena_val" ]]; then
+        echo "options irdma roce_ena=1" | sudo tee -a /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    elif [[ "$roce_ena_val" != "1" ]]; then
+        sed -i '/options irdma roce_ena=/s/roce_ena=[0-9]*/roce_ena=1/' /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    fi
+
+    # increase irdma Queue Pair limit
+    limits_sel_val=$(grep "options irdma limits_sel=" /etc/modprobe.d/irdma.conf | cut -d "=" -f 2)
+    if [[ -z "$limits_sel_val" ]]; then
+        echo "options irdma limits_sel=5" | sudo tee -a /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    elif [[ "$limits_sel_val" != "5" ]]; then
+        sed -i '/options irdma limits_sel=/s/limits_sel=[0-9]*/limits_sel=5/' /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    fi
+}
+
+install_perftest() {
+    echo "Installing perftest dependencies..."
+    sudo apt update
+    sudo apt install -y libibverbs-dev librdmacm-dev libibumad-dev libpci-dev
+    echo "Installing perftest..."
+    git clone https://github.com/linux-rdma/perftest.git "$WORKING_DIR"/perftest
+    pushd "$WORKING_DIR"/perftest
+    ./autogen.sh
+    ./configure
+    make
+    sudo make install
+    popd
+}
+
+check_roce() {
+    echo -n "Checking RoCE... "
+    roce_ena_val=$(cat /sys/module/irdma/parameters/roce_ena)
+    if [[ "$roce_ena_val" != "1" ]] && [[ "$roce_ena_val" != "65535" ]]; then
+        echo "FAIL: RoCE disabled"
+        exit 1
+    fi
+    echo "OK"
+}
+
+check_mtu() {
+    echo "Checking MTU..."
+    rdma_devices=($(rdma link | grep "LINK_UP" | cut -d " " -f 8))
+    if [[ "${#rdma_devices[@]}" == "0" ]]; then
+        echo "FAIL: No RDMA devices detected"
+        exit 1
+    fi
+    fail=0
+    for device in "${rdma_devices[@]}"; do
+        mtu=$(ip addr | grep "$device:" | cut -d " " -f 5)
+        if [[ "$mtu" == "9000" ]]; then
+            echo "* $device OK"
+        else
+            echo "* $device FAIL: MTU = $mtu (9000 is recommended)"
+            fail=1
+        fi
+    done
+    if [[ $fail -eq 1 ]]; then
+        exit 1
+    fi
+}
+
+set_mtu() {
+    echo -n "Setting MTU to 9000 on $1... "
+    sudo ip link set dev $1 mtu 9000
+    echo "DONE"
+}
+
+run_perftest() {
+    sudo ip addr add 192.168.255.255/24 dev $1
+    taskset -c 1 ib_write_bw --qp=4 --report_gbit -D 60 --tos 96 -R &> "$WORKING_DIR"/perftest_server.log &
+    server_pid=$!
+    ib_write_bw 192.168.255.255 --qp=4 --report_gbit -D 60 --tos 96 -R &> "$WORKING_DIR"/perftest_client.log &
+    client_pid=$!
+    echo "perftest is running. Waiting 60 s..."
+    wait $server_pid
+    wait $client_pid
+    echo "perftest completed. See results in $WORKING_DIR/perftest_server.log and $WORKING_DIR/perftest_client.log"
+    sudo ip addr del 192.168.255.255/24 dev $1
+}
```

Comment:
No need to change the execution part, as there are defaults (except `interface_name`)
```suggestion
run_perftest() {
    # Usage: run_perftest <interface_name> [network_address] [network_mask]
    interface_name="$1"
    network_address="${2:-192.168.255.255}"
    network_mask="${3:-24}"
    sudo ip addr add ${network_address}/${network_mask} dev ${interface_name}
    taskset -c 1 ib_write_bw --qp=4 --report_gbit -D 60 --tos 96 -R &> "$WORKING_DIR"/perftest_server.log &
    server_pid=$!
    ib_write_bw ${network_address} --qp=4 --report_gbit -D 60 --tos 96 -R &> "$WORKING_DIR"/perftest_client.log &
    client_pid=$!
    echo "perftest is running. Waiting 60 s..."
    wait $server_pid
    wait $client_pid
    echo "perftest completed. See results in $WORKING_DIR/perftest_server.log and $WORKING_DIR/perftest_client.log"
    sudo ip addr del ${network_address}/${network_mask} dev ${interface_name}
}
```